docs/docsite/rst/lookup_guide.rst

```diff
@@ -0,0 +1,159 @@
+.. _ansible_collections.community.hashi_vault.docsite.lookup_guide:
+
+************
+Lookup guide
+************
+
+This guide is not a comprehensive listing of included lookup plugins and how to use them, rather it is intended to explain the role of the lookup plugins in this collection and how they are they used, especially when compared to modules of the same name.
+
+.. contents::
+  :local:
+  :depth: 2
+
+
+About the ``hashi_vault`` lookup
+================================
+
+Before this collection existed, the ``hashi_vault`` lookup plugin was the only Vault-related content included in Ansible. Eventually it was moved to :ref:`community.general <plugins_in_community.general>` before being split out into this collection, where it was the only content in the collection for a long time. As a result, it's the most used plugin for Vault, and the one most people are familiar with.
+
+``hashi_vault`` lookup considerations
+-------------------------------------
+
+Due to the history of the ``hashi_vault`` lookup's development, and it being the only one around, it kind of does many jobs, making it versatile, but sometimes unintuitive.
+
+First, the lookup does authentication, taking parameters for various login types, performing a login, and acquiring a token with which it can make additional calls to Vault.
+
+Second, the lookup does a generic read. That makes it very useful because it can read any kind of Vault path, without having to be written with that type of path in mind.
+
+It's worth understanding that by far the most common use case is reading secrets, and by far the most common secret store is the ``kv`` (key/value) store that's built into Vault. Most implementations are using v2 of the ``kv`` store.
+
+So the third thing the lookup does is assume that you're probably trying to read a ``kv`` secret, and to then try to infer if the response is from ``kv2``, because the responses from version 2 include metadata and have the secret value additionally wrappped in another structure. In that way, it seeks to make the response from version 2 look more like the response from version 1.
+
+Since the ``kv`` store has one or more key/value pairs in each secret, the lookup also supports a non-standard suffix in its path that can be used to access a value belonging to one specific key, via the ``:keyname`` syntax. While this is useful to provide a compact way to access a single secret value (admittedly a very common use case), it complicates the implementation and leads to bad habits.
+
+For example, it became common to see people use many lookup invocations with the same path, each with a different ``:keyname``, to access multiple values within a single secret, but this is quite wasteful, as it does a separate login and secret lookup, all to return the same value, and the key dereferencing is done client side. Further, dereferencing can be done directly in Jinja where it's more clear what's going on, using the ``.key`` or ``['key']`` syntax.
```

Comment:
An example would be very helpful here, especially a "before and after" type of example, showing the same series of lookups, first done the "bad habit" way, then done the better way.

Reply:
I actually do intend to move a bunch of this to a new page in the future, but I wanted to put some of it down somewhere for now.

What I'm waiting for for the other page, is the expected `vault_kv_` series of plugins and modules, which together with `vault_read` will provide full replacement coverage for `hashi_vault`, by separating the functionality into "reading kv secrets" and "reading anything else".

At that point, I'd have a guide specifically for migrating away from using `hashi_vault` and what to use in its place, that would be full of examples and such.

---

So this paragraph, could maybe be shortened or removed entirely (I don't necessarily need to explain what the bad habits are here).